Formula/dotnet.rb

```diff
@@ -0,0 +1,88 @@
+class Dotnet < Formula
+  desc ".Net Core"
+  homepage "https://github.com/dotnet/cli/releases"
+  url "https://github.com/dotnet/cli/archive/v1.0.0-preview2.tar.gz"
+  sha256 "a0708ed3eedb10e95dcea96c62f85be1c38b91f2be9afa89d8fc655b7e0c4706"
+
+  patch :DATA
+  head "https://github.com/dotnet/cli.git"
+
+  depends_on "cmake" => :build
+  depends_on "openssl"
+
+  def install
+    system "./build.sh", "--targets", "Prepare,Compile"
+
+    stage2 = buildpath/"artifacts/osx.#{MacOS.version}-x64/stage2"
+    # TODO: how do I tell homebrew that the folders are under stage2?
+
+    # Relink openssl to the homebrew version
+    # Would be smoother to tell clang to use this openssl prefix, but alas...
+    openssl_path = Formula["openssl"].opt_prefix
+    sh "install_name_tool -add_rpath #{openssl_path}/lib #{stage2}/shared/Microsoft.NETCore.App/1.0.0/System.Security.Cryptography.Native.dylib"
+
+    bin.install stage2/"dotnet"
+    prefix.install Dir["*"]
```

Comment:
This might need to be `prefix.install Dir["#{buildpath}/artifacts/osx.#{MacOS.version}-x64/stage2/*"]`, but if everything other than the `dotnet` binary belongs in the root prefix (as opposed to `lib`, `libexec`, etc.) then yeah, this is the right approach.